net: config: Fix improper condition in net_config_init

lmaciejonczyk opened the pull request:
Currently, there is a case for net_config_init function that for timeout==0 and when iface is already up, the ip setup is not proceed
and the error message "Timeout while waiting network..." is logged.

This PR fixes it.

pfalcon commented:
> net_config_init returns although network interface may be up for a case when timeout is set to 0. This PR fixes it.

@lmaciejonczyk: Can you please rewrite and extend the commit message, so it was clear what was the problem, and how this patch fixes it? I for example not sure I get enough info from the phrase above, and so not sure that the changes do exactly what was described. Thanks.

lmaciejonczyk commented:
> > net_config_init returns although network interface may be up for a case when timeout is set to 0. This PR fixes it.
> 
> @lmaciejonczyk: Can you please rewrite and extend the commit message, so it was clear what was the problem, and how this patch fixes it? I for example not sure I get enough info from the phrase above, and so not sure that the changes do exactly what was described. Thanks.

Sure, no problem. Is it better now?

pfalcon commented:
@lmaciejonczyk, Much better, and good catch on that issue, thanks! (I'll review in more detail a bit later, unless other colleagues will be first.)

lmaciejonczyk committed:
net: config: Fix improper condition in net_config_init

Currently, there is a case for net_config_init function that for
timeout==0 and when iface is already up, the ip setup is not proceed
and the error message "Timeout while waiting network..." is logged.

This commit fixes it.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

jukkar commented:
This looks like a valid issue. @lmaciejonczyk could you rebase against latest master and force push a new version.
